### ISSUE TITLE
Conistent singular and plural use for TensorBlocks in the documentation and error messages

### ISSUE THREAD

**Issue by agoscinski** [lab-cosmo/equistore#217]
It seems a bit pedantic, but TensorBlocks are complicated objects and having sometimes a plural in a variable name where it should not be, makes the understanding harder. I don't think we have to be consistent everywhere, but it would be good if we are at least consistent in the documentation and error messages. To give concrete examples in
https://lab-cosmo.github.io/equistore/latest/reference/python/operations/manipulation/samples-reduction.html#equistore.sum_over_samples
we are using the singular in samples Labels names `structure` and `center`, but for the properties we use the plural `properties`. This is not so confusing here, but I let us look at an example from the tests.
https://github.com/lab-cosmo/equistore/blob/c3e8f700e6906c043211983ad7e4e1b865081a33/python/tests/utils.py#L6
https://github.com/lab-cosmo/equistore/blob/c3e8f700e6906c043211983ad7e4e1b865081a33/python/tests/utils.py#L28
https://github.com/lab-cosmo/equistore/blob/c3e8f700e6906c043211983ad7e4e1b865081a33/python/tests/utils.py#L35-L38
Here we use the plural in the values and the singular in the gradients. Since we enforce the singular for the name of the gradient block samples Labels (otherwise one gets an error), I would say that the values also should use the singular.

Sometimes in the code we use the term gradient**s** to refer to the values of a gradient block, sometimes we refer to the different gradient block**s**. Since at least the python API is using gradient**s** to refer to gradient blocks, I would stick with this usage.

**Comment by ceriottm** [comment]
I agree on the need for consistency, but it is IMO correct to have the `samples` and `properties` members to be plural, and the name of the columns be e.g. `["sample", "structure"]`. In the first case, we refer to a list of samples, in the second it's like the title of the column that indicates that the numbers below indicate the indices of the sample and of the structure each row corresponds to

**Comment by agoscinski** [comment]
I agree on this. I think I wasnt so clear in my text. What I would change in this example below is the Labels names (I think you refer to it with "name of the columns")
```python
    block_1 = TensorBlock(
        values=np.full((3, 1, 1), 1.0),
        samples=Labels(["samples"], np.array([[0], [2], [4]], dtype=np.int32)),
        components=[Labels(["components"], np.array([[0]], dtype=np.int32))],
        properties=Labels(["properties"], np.array([[0]], dtype=np.int32)),
    )
    block_1.add_gradient(
        "parameter",
        samples=Labels(
            ["sample", "parameter"], np.array([[0, -2], [2, 3]], dtype=np.int32)
        ),
        data=np.full((2, 1, 1), 11.0),
        components=[Labels(["components"], np.array([[0]], dtype=np.int32))],
    )
```
so this would be become
```python
    block_1 = TensorBlock(
        values=np.full((3, 1, 1), 1.0),
        samples=Labels(["sample"], np.array([[0], [2], [4]], dtype=np.int32)),
        components=[Labels(["component"], np.array([[0]], dtype=np.int32))],
        properties=Labels(["property"], np.array([[0]], dtype=np.int32)),
    )
    block_1.add_gradient(
        "parameter",
        samples=Labels(
            ["sample", "parameter"], np.array([[0, -2], [2, 3]], dtype=np.int32)
        ),
        data=np.full((2, 1, 1), 11.0),
        components=[Labels(["component"], np.array([[0]], dtype=np.int32))],
    )
```

**Comment by ceriottm** [comment]
OK yes then I agree. We could even call the columns something like `sample_idx` and `property_idx` for additional clarity
